### PR TITLE
mpiext/pcollreq: check subroutine parameters and add profiling symbols

### PR DESCRIPTION
**Pull request by ggouaillardet**
- check subroutine parameters
 - implement PMPIX_* subroutines

Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>

### CODE REVIEW

**Comment by ibm-ompi** [comment]
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/4cf9d810d06d4a0e4b2e80ce7c40847e

**Comment by ibm-ompi** [comment]
The IBM CI (XL Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/936c21da601ff43091a5ddd098daffb6

**Comment by ibm-ompi** [comment]
The IBM CI (PGI Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/76f750f16dbfd6052f9fecb20dbc8278

**Comment by ibm-ompi** [comment]
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/a3ac1518b40a5a9b257ff86d27b90282

**Comment by ggouaillardet** [comment]
@kawashima-fj can you please have a look before I merge (ideally, before the `v4` branch) ?

File: ompi/mpiext/pcollreq/c/alltoall_init.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#include "ompi/runtime/ompi_spc.h"
+
+#if OMPI_BUILD_MPI_PROFILING
+#if OPAL_HAVE_WEAK_SYMBOLS
+#pragma weak MPIX_Alltoall_init = PMPIX_Alltoall_init
+#endif
+#define MPIX_Alltoall_init PMPIX_Alltoall_init
+#endif
+
+static const char FUNC_NAME[] = "MPI_Alltoall_init";
```
Comment (kawashima-fj) [inline ompi/mpiext/pcollreq/c/alltoall_init.c]: `MPI_Alltoall_init` -> `MPIX_Alltoall_init` (`X` missing)

**Comment by jsquyres** [comment]
This PR also includes an SPC commit.  Is that what you intended?  It's really unrelated to the mpiext work.

**Comment by ggouaillardet** [comment]
@kawashima-fj thanks a lot for the review, I just updated my PR to fix the issues you spotted.

@jsquyres most files are copied from the C bindings (for example `ompi/mpiext/pcollreq/c/bcast_init.c` was copied from `ompi/mpi/c/ibcast.c`) and then adapted.
When doing this, I noted some things related to SPC were missing here and there in the C bindings, so I fixed them and it ended in a different commint but in the same PR.

I can split that in two PR if needed, but they will have to be merged in a specific order.
imho, the SPC changes are trivial enough and hence do not require such extra pain.

**Pull request merged**